code/modules/library/lib_items.dm

```diff
@@ -61,7 +61,8 @@
 				state = 0
 
 		if(2)
-			if(istype(I, /obj/item/weapon/book) || istype(I, /obj/item/weapon/spellbook))
+			var/list/allowed_books = list(/obj/item/weapon/book, /obj/item/weapon/spellbook, /obj/item/weapon/storage/book)

```

Comment:
Yeah, it does, considering the variable is only defined in an 'if' statement. That might cause some issues, gonna remedy that